components/script/dom/urlhelper.rs

```diff
@@ -149,6 +153,7 @@ impl UrlHelper {
     }
 
     pub fn SetSearch(url: &mut Url, value: USVString) {
+        url.query = Some(String::new());
```

Comment:
This implements step 5.2 from the spec for the search setter: https://html.spec.whatwg.org/multipage/semantics.html#dom-hyperlink-search . From what I've seen, all the uses of `UrlHelper` have a similar spec, so it made sense to put this here rather than, e.g., in `htmlanchorelement.rs`. Same for the other changes.

@jdm and I discussed this a little earlier in this PR thread.